app/src/main/java/org/oppia/app/walkthrough/WalkthroughActivityPresenter.kt

```diff
@@ -1,25 +1,84 @@
 package org.oppia.app.walkthrough
 
 import androidx.appcompat.app.AppCompatActivity
+import androidx.databinding.DataBindingUtil
 import org.oppia.app.R
 import org.oppia.app.activity.ActivityScope
+import org.oppia.app.databinding.WalkthroughActivityBinding
+import org.oppia.app.viewmodel.ViewModelProvider
+import org.oppia.app.walkthrough.end.WalkthroughFinalFragment
+import org.oppia.app.walkthrough.topiclist.WalkthroughTopicListFragment
+import org.oppia.app.walkthrough.welcome.WalkthroughWelcomeFragment
 import javax.inject.Inject
 
 /** The presenter for [WalkthroughActivity]. */
 @ActivityScope
-class WalkthroughActivityPresenter @Inject constructor(private val activity: AppCompatActivity) {
+class WalkthroughActivityPresenter @Inject constructor(
+  private val activity: AppCompatActivity,
+  private val viewModelProvider: ViewModelProvider<WalkthroughViewModel>
+) {
+  private lateinit var binding: WalkthroughActivityBinding
 
   fun handleOnCreate() {
-    activity.setContentView(R.layout.walkthrough_activity)
-    if (getWalkthroughFragment() == null) {
+    binding = DataBindingUtil.setContentView(activity, R.layout.walkthrough_activity)
+
+    binding.apply {
+      viewModel = getWalkthroughViewModel()
+      presenter = this@WalkthroughActivityPresenter
+      lifecycleOwner = activity
+    }
+    if (getWalkthroughWelcomeFragment() == null) {
       activity.supportFragmentManager.beginTransaction().add(
         R.id.walkthrough_fragment_placeholder,
-        WalkthroughFragment()
-      ).commitNow()
+        WalkthroughWelcomeFragment()
+      ).commitNow().also {
+        getWalkthroughViewModel().currentProgress.set(1)
+      }
+    }
+  }
+
+  fun prevPage(pageNo: Int) {
```

Comment:
rename prevPage to `previousPage`

Replace pageNo to walkthroughPage because directly we are passing a WalkthroughPages item.


In short do not use shorthand names like prev, No. etc.

Reply:
done